worlds/mmx/__init__.py

```diff
@@ -0,0 +1,402 @@
+    climb walls, which were new to the series at the time.
+    """
+    game = "Mega Man X"
+    web = MMXWeb()
+
+    settings: ClassVar[MMXSettings]
+    
+    options_dataclass = MMXOptions
+    options: MMXOptions
+    
+    required_client_version = (0, 5, 0)
+
+    item_name_to_id = {name: data.code for name, data in item_table.items()}
+    location_name_to_id = all_locations
+    item_name_groups = item_groups
+    location_name_groups = location_groups
+    hint_blacklist = {
+        LocationName.armored_armadillo_clear,
+        LocationName.chill_penguin_clear,
+        LocationName.boomer_kuwanger_clear,
+        LocationName.sting_chameleon_clear,
+        LocationName.storm_eagle_clear,
+        LocationName.flame_mammoth_clear,
+        LocationName.spark_mandrill_clear,
+        LocationName.launch_octopus_clear,
+        LocationName.intro_completed,
+    }
+
+    def __init__(self, multiworld: MultiWorld, player: int) -> None:
+        self.rom_name_available_event = threading.Event()
+        super().__init__(multiworld, player)
+
+    def create_regions(self) -> None:
+        location_table = setup_locations(self)
+        create_regions(self, location_table)
+
+        itempool: List[MMXItem] = []
+        
+        connect_regions(self)
+        
+        total_required_locations = 47
+        if self.options.pickupsanity:
+            total_required_locations += 26
+
+        # Add levels into the pool
+        start_inventory = self.options.start_inventory.value.copy()
+        stage_list = [
+            ItemName.stage_armored_armadillo,
+            ItemName.stage_boomer_kuwanger,
+            ItemName.stage_chill_penguin,
+            ItemName.stage_flame_mammoth,
+            ItemName.stage_launch_octopus,
+            ItemName.stage_spark_mandrill,
+            ItemName.stage_sting_chameleon,
+            ItemName.stage_storm_eagle,
+        ]
+        stage_selected = self.random.randint(0, 7)
+        if any(stage in self.options.start_inventory_from_pool for stage in stage_list) or \
+           any(stage in start_inventory for stage in stage_list):
+            total_required_locations += 1
+            for i in range(len(stage_list)):
+                if stage_list[i] not in start_inventory:
+                    itempool += [self.create_item(stage_list[i])]
+        else:
+            for i in range(len(stage_list)):
+                if i == stage_selected:
+                    self.multiworld.get_location(LocationName.intro_completed, self.player).place_locked_item(self.create_item(stage_list[i]))
+                else:
+                    itempool += [self.create_item(stage_list[i])]
+
+        if len(self.options.sigma_open.value) == 0:
+            itempool += [self.create_item(ItemName.stage_sigma_fortress)]
+
+        # Add weapons into the pool
+        itempool += [self.create_item(ItemName.electric_spark)]
```

Comment:
I think all of these `itempool += [` would be better as `itempool.append(` instead.
That way it doesn't create a new list for every `+=`